### PR TITLE
[MIG] beesdoo_purchase: migrate to 12.0

### PR DESCRIPTION
**Pull request by sdegueldre**
This is a work in progress, the branch is not ready to be merged. Creating this so that others can see it's already being worked on.

### CODE REVIEW

**Comment by remytms** [comment]
@sdegueldre I've rebased `12.0-mig-obeesdoo` the branch that will become our new `12.0`.
I've also rebased your branch `12.0-mig-beesdoo-purchase` on `12.0-mig-obeesdoo`. Can you check that I've not deleted an important commit. Your precedent work has been saved on the branch `12.0-mig-beesdoo-purchase-save-2020-01-17`.

Thanks for your work. :)

**Comment by mclaeysb** [comment]
More info on `message_subscribe_users()` [here](https://github.com/odoo/odoo/commit/b6d2351df45bdd049e42e828743b5dd2f1941649)

@remytms could you explain the domain that is added on the products? What is it for, what doest the `parent` refer to?

File: beesdoo_purchase/__manifest__.py [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+        Extends Purchase module. 
+        """,
+    "description": """
+        Extends Purchase module: 
+        - Adds a 'Responsible' field to purchase orders:
+        This is a user who will follow up the order. This users replaces 
+        the creator in the order's mail messages followers list, and in the 
+        create_uid ORM field. His user's contact info is printed on 
+        purchases orders as 'Referent'.
+        - A filter w.r.t. the mail sellers is placed on the products field of a
```
Comment (vvrossem) [inline beesdoo_purchase/__manifest__.py:13]: TIL what wrt means :partying_face:

**Review by vvrossem — COMMENTED** [review]
LGTM.
I've read you question to @remytms: [I think it gets the `partner_id` in the parent form ](https://stackoverflow.com/questions/31244937/how-to-get-a-field-of-the-parent-form-from-a-pop-up-in-odoo) but no idea why

**Review by vvrossem — APPROVED** [review]

**Pull request merged**